packages/@lwc/ssr-compiler/src/compile-js/wire.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { is, builders as b } from 'estree-toolkit';
+import { isIdentifierName, isReservedWord } from '@babel/helper-validator-identifier';
+import { esTemplate } from '../estemplate';
+
+import type { PropertyDefinition as EsPropertyDefinitionWithDecorators } from 'meriyah/dist/src/estree';
+import type {
+    Expression as EsExpression,
+    PropertyDefinition as EsPropertyDefinition,
+    ObjectExpression as EsObjectExpression,
+    Statement as EsStatement,
+} from 'estree';
+import type { ComponentMetaState, WireAdapter } from './types';
+
+function extractWireConfig(
+    propName: string,
+    adapterConstructorId: string,
+    config?: EsObjectExpression
+): WireAdapter {
+    const extractedConfig =
+        config?.properties?.map?.((objProp) => {
+            if (!is.property(objProp)) {
+                throw new Error('Object spread syntax is disallowed in @wire config.');
+            }
+            const { key, value } = objProp;
+
+            if (!is.identifier(key)) {
+                throw new Error(`@wire config entry key must be an identifer; found ${key.type}`);
+            }
```

Comment:
We support computed properties with literals now:

https://github.com/salesforce/lwc/blob/4845840a8f11a0f0334a7fb50dd82b4f7c2cbe22/packages/%40lwc/babel-plugin-component/src/__tests__/fixtures/wire-decorator/transforms-computed-properties/actual.js#L11-L17

Reply:
I'll open a separate WI for this.

Reply:
https://github.com/salesforce/lwc/issues/4827